src/YieldDisburser.sol

```diff
@@ -10,60 +10,115 @@ import {Bread} from "../lib/bread-token-v2/src/Bread.sol";
-    error InvalidSignature();
+    // @notice The error emitted when attempting to instantiate a variable with a zero value
     error MustBeGreaterThanZero();
+    // @notice The error emitted when attempting to vote with a point value greater than pointsMax
     error VotePointsTooLarge();
+    // @notice The error emitted when a voter has never held bread before
     error NoCheckpointsForAccount();
+    // @notice The error emitted when attempting to calculate voting power for a period with a start block greater than the end block
     error StartMustBeBeforeEnd();
+    // @notice The error emitted when attempting to distribute yield when access conditions are not met
     error YieldNotResolved();
+    // @notice The error emitted when attempting to distribute yield with a balance less than the number of projects
     error YieldTooLow(uint256);
+    // @notice The error emitted when attempting to remove a project that is not in the breadchainProjects array
     error ProjectNotFound();
+    // @notice The error emitted when attempting to add or remove a project that is already queued for addition or removal
     error ProjectAlreadyQueued();
+    // @notice The error emitted when attempting to add a project that is already in the breadchainProjects array
     error AlreadyMemberProject();
+    // @notice The error emitted when a user attempts to vote without the minimum required voting power
     error BelowMinRequiredVotingPower();
 
     /// @custom:oz-upgrades-unsafe-allow constructor
     constructor() {
         _disableInitializers();
     }
 
-    function initialize(address breadAddress, address[] memory _breadchainProjects) public initializer {
+    function initialize(
+        address breadAddress,
+        address[] memory _breadchainProjects,
+        uint256 _blocktime,
+        uint256 _minVotingAmount,
+        uint256 _minVotingHoldingDuration,
+        uint256 _maxVotes,
+        uint256 _pointsMax,
+        uint48 _minimumTimeBetweenClaims,
+        uint48 _lastClaimedTimestamp,
+        uint256 _lastClaimedBlocknumber,
+        uint256 _precision
+    ) public initializer {
         breadToken = Bread(breadAddress);
         breadchainProjects = new address[](_breadchainProjects.length);
         for (uint256 i; i < _breadchainProjects.length; ++i) {
             breadchainProjects[i] = _breadchainProjects[i];
         }
-        minVotingAmount = 5; // must hold atleast 5 bread
-        minVotingHoldingDuration = 10 days; // must hold for atleast 10 days
-        minRequiredVotingPower = 1e18 * minVotingAmount * minVotingHoldingDuration / BLOCKTIME; // Holding 10 bread for 5 days , assuming a 5 second block time
-        maxVotes = 1e4;
-        pointsMax = 100000;
+        BLOCKTIME = _blocktime;
+        PRECISION = _precision;
+        minVotingAmount = _minVotingAmount;
+        minVotingHoldingDuration = _minVotingHoldingDuration * 1 days; // must hold for atleast _minVotingHoldingDuration  days
```

Comment:
i would advocate for not doing this calculation and just assume the deployer knows to set the time in seconds instead of days

Reply:
Why? this hurts readability and would be inconsistent with the setter.